### PR TITLE
[MoE][3/n] consolidate EP=1 and EP>1 to all use All2AllTokenDispatcher

### PR DESCRIPTION
**Pull request by acisseJZhong**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #3125
* * __->__ #2960

## Summary

Unify EP=1 and EP>1 token dispatching by making `AllToAllTokenDispatcher` fall back to `LocalTokenDispatcher` behavior when `ep_mesh is None`.

### Changes

- **`AllToAllTokenDispatcher.dispatch/combine`**: When `ep_mesh is None` (EP=1), falls back to `super().dispatch()`/`super().combine()` (LocalTokenDispatcher) — no all-to-all communication, just local token reordering.
- **`make_token_dispatcher_config`**: Default `comm_backend` changed from `None` to `"standard"`. Removed the `None` → `LocalTokenDispatcher` path. All MoE models now use `AllToAllTokenDispatcher` by default.
- **All model `__init__.py` files**: `moe_comm_backend` default changed from `str | None = None` to `str = "standard"`.
- **All `model_registry` functions**: Removed `if moe_comm_backend is not None` guards — `moe_comm_backend` is always passed through directly.
- **All config_registry files**: Removed redundant `moe_comm_backend="standard"` since it's now the default.

### Why

Previously, users had to manually set `moe_comm_backend=None` when overriding `expert_parallel_degree=1` on the command line, otherwise the `AllToAllTokenDispatcher` would assert `ep_group is not None`(see issue in https://github.com/pytorch/torchtitan/issues/3082,). Now the dispatcher handles EP=1 gracefully by falling back to local dispatch.

### Test command:
```
NCCL_NVLS_ENABLE=0 torchrun --nnodes 1 --nproc-per-node 8 -m torchtitan.train --module deepseek_v3 --config deepseek_v3_16b --parallelism.tensor_parallel_degree 8 --parallelism.context_parallel_degree 1 --parallelism.expert_parallel_degree 1 --training.steps 10 --dataloader.dataset c4_test
```

### CODE REVIEW

File: 1
Comment (tianyu-l) [inline 1:1]: remove?

File: torchtitan/experiments/graph_trainer/llama3/__init__.py [hunk 1]
```diff
@@ -18,12 +18,11 @@
 def model_registry(
     flavor: str,
     attn_backend: str = "sdpa",
-    moe_comm_backend: str | None = None,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/experiments/graph_trainer/llama3/__init__.py]: non moe model shouldn't have this field

File: torchtitan/experiments/graph_trainer/qwen3/__init__.py [hunk 2]
```diff
@@ -18,12 +18,11 @@
 def model_registry(
     flavor: str,
     attn_backend: str = "sdpa",
-    moe_comm_backend: str | None = None,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/experiments/graph_trainer/qwen3/__init__.py]: maybe need to be optional for qwen3, as it has non moe variants

we can assert its existence inside qwen3 moe variants

File: torchtitan/models/common/config_utils.py [hunk 3]
```diff
@@ -199,23 +198,17 @@ def make_token_dispatcher_config(
     num_experts: int,
     top_k: int,
     score_before_experts: bool = True,
-    comm_backend: str | None = None,
+    comm_backend: str = "standard",
     non_blocking_capacity_factor: float | None = None,
-) -> LocalTokenDispatcher.Config:
+) -> AllToAllTokenDispatcher.Config:
```
Comment (tianyu-l) [inline torchtitan/models/common/config_utils.py]: return type should be common base class, or union of classes

File: torchtitan/models/common/token_dispatcher.py [hunk 4]
```diff
@@ -451,14 +455,15 @@ def combine(
...
+    """Token dispatcher with token group padding for quantized grouped GEMMs.
 
     Uses torchao's ``permute_and_pad`` instead of the standard ``_permute`` to
     reorder tokens into expert-major order and pad each expert's token group to
     a multiple of ``pad_multiple``. This alignment is required by FP8/MXFP8
     quantized grouped GEMM kernels (e.g. 16 for FP8, 32 for MXFP8).
 
-    ep_mesh is set by ExpertParallel / ExpertTensorParallel._apply().
+    When EP is enabled (ep_mesh is set), uses all-to-all communication.
+    When EP=1 (ep_mesh is None), falls back to local dispatch.
```
Comment (tianyu-l) [inline torchtitan/models/common/token_dispatcher.py]: When EP=1, there's no permute / unpermute, so I don't think it works with quantized grouped gemm. We just need to error out somewhere properly.

File: torchtitan/models/deepseek_v3/__init__.py [hunk 5]
```diff
@@ -171,7 +171,7 @@ def _build_dsv3_layers(
     router_route_norm: bool = False,
     score_before_experts: bool = False,
     attn_backend: str,
-    moe_comm_backend: str | None = None,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/models/deepseek_v3/__init__.py]: don't need default here?

File: torchtitan/models/deepseek_v3/__init__.py [hunk 6]
```diff
@@ -447,7 +447,7 @@ def _236b(
 
 def _671b(
     attn_backend: str = "flex",
-    moe_comm_backend: str | None = None,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/models/deepseek_v3/__init__.py]: also, don't need these defaults, as they are always given by `model_registry`

File: torchtitan/models/gpt_oss/__init__.py [hunk 7]
```diff
@@ -124,7 +124,7 @@ def _make_gptoss_experts_config(
     layer_id: int,
     top_k: int,
     score_before_experts: bool,
-    moe_comm_backend: str | None = None,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/models/gpt_oss/__init__.py]: also don't need default

File: torchtitan/models/llama3/__init__.py [hunk 8]
```diff
@@ -104,7 +104,7 @@ def _build_llama3_layers(
     return layers
 
 
-def _debugmodel(attn_backend: str = "sdpa") -> Llama3Model.Config:
+def _debugmodel(attn_backend: str = "sdpa", moe_comm_backend: str = "standard") -> Llama3Model.Config:
```
Comment (tianyu-l) [inline torchtitan/models/llama3/__init__.py]: there shouldn't be moe backend for dense models

File: 1
Comment (acisseJZhong) [reply 1:1]: oops claude is doing sth weird. I removed it once, but seems everytime I commit, it re appears :(

File: torchtitan/experiments/graph_trainer/qwen3/__init__.py (hunk 2 above)
Comment (acisseJZhong) [reply torchtitan/experiments/graph_trainer/qwen3/__init__.py]: restore this back to `moe_comm_backend: str | None = None`; and only MoE model builders will have this `moe_comm_backend` which defaults to `standard`.

File: torchtitan/models/common/token_dispatcher.py (hunk 4 above)
Comment (acisseJZhong) [reply torchtitan/models/common/token_dispatcher.py]: if we want to support quantization for MoE EP=1, I believe it's possible if we set padding needs to be handled by torchao? However I think per last discussion, we would prefer banning EP=1 with torchao dispatcher? I will error out for now. cc @danielvegamyhre

Comment (danielvegamyhre) [reply torchtitan/models/common/token_dispatcher.py]: Yeah that sounds good to me

File: torchtitan/models/deepseek_v3/__init__.py (hunk 6 above)
Comment (acisseJZhong) [reply torchtitan/models/deepseek_v3/__init__.py]: rethinking this, maybe we should get rid of the default here for both attn_backend and moe_comm_backend as both are provided through model_registry.

File: torchtitan/models/common/config_utils.py [hunk 9]
```diff
@@ -199,23 +199,17 @@ def make_token_dispatcher_config(
     num_experts: int,
     top_k: int,
     score_before_experts: bool = True,
-    comm_backend: str | None = None,
+    comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/models/common/config_utils.py]: model_registry should be the only function that has default for this field? o/w when would the default be used here?

File: torchtitan/models/qwen3_vl/__init__.py [hunk 10]
```diff
@@ -283,7 +289,8 @@ def _debugmodel() -> Qwen3VLModel.Config:
 
 
 def _debugmodel_moe(
-    moe_comm_backend: str | None = None,
+    attn_backend: str,
+    moe_comm_backend: str = "standard",
```
Comment (tianyu-l) [inline torchtitan/models/qwen3_vl/__init__.py:293]: similar

**Review by tianyu-l — APPROVED** [review]

**Comment by acisseJZhong** [comment]
failing test not caused by this PR, landing
